api/base/serializers.py

```diff
@@ -13,6 +12,27 @@
 from api.base.exceptions import InvalidQueryStringError, Conflict
 
 
+def format_relationship_links(related_link, self_link, rel_meta, self_meta):
+
+    ret = {
+        'links': {
+            'related': {
+                'href': related_link,
+                'meta': rel_meta
+            },
+            'self': {
+                'href': self_link,
+                'meta': self_meta
+            }
+        }
+    }
+
+    # Not currently returning self links
+    if not ret['links']['self']['href']:
```

Comment:
It seems roundabout to check the dictionary you created `ret['links']['related']['href']` (rather than checking the input value, `self_link`, directly) and then delete the key from the dictionary.

It seems roundabout to add the `'related'` key, check the '[related']['href']`, then delete `'related'. Why not conditionally add `'related'`?